File: src/main/kotlin/com/elissamc/actions/JavaArbitraryAction.java
package com.elissamc.actions;

import cn.nukkit.Player;
import com.elissamc.util.ParametrizedRunnable;

/**
 * Menu action that will execute specified code, when triggered.
 */
public class JavaArbitraryAction implements Action {
    private final ParametrizedRunnable runnable;

    public JavaArbitraryAction(final ParametrizedRunnable runnable) {
        this.runnable = runnable;
    }

    @Override
    public void execute(final Player player) {
        this.runnable.run(player);
    }
}

File: src/main/kotlin/com/elissamc/actions/JavaArbitraryAction.kt
package com.elissamc.actions

import cn.nukkit.Player
import com.elissamc.util.ParametrizedRunnable

/**
 * Menu action that will execute specified code, when triggered.
 */
class JavaArbitraryAction(private val runnable: ParametrizedRunnable) : Action {

    override fun execute(player: Player) {
        this.runnable.run(player)
    }
}